modules/apps/commerce/headless/headless-commerce-core-api/src/main/java/com/liferay/headless/commerce/core/util/CurrencyUtil.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.headless.commerce.core.util;
+
+import com.liferay.commerce.currency.exception.NoSuchCurrencyException;
+import com.liferay.commerce.currency.model.CommerceCurrency;
+import com.liferay.commerce.currency.service.CommerceCurrencyLocalServiceUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+/**
+ * @author Luca Pellizzon
+ */
+public class CurrencyUtil {
+
+	public static CommerceCurrency getCommerceCurrency(
+			long companyId, String currencyCode,
+			String currencyExternalReferenceCode, long currencyId)
+		throws NoSuchCurrencyException {
+
+		CommerceCurrency commerceCurrency = null;
+
+		if (Validator.isNull(currencyCode)) {
+			commerceCurrency =
+				CommerceCurrencyLocalServiceUtil.fetchCommerceCurrency(
+					companyId, currencyCode);
+
+			if (commerceCurrency != null) {
+				return commerceCurrency;
+			}
+		}
+
+		if (currencyId > 0) {
+			commerceCurrency =
+				CommerceCurrencyLocalServiceUtil.fetchCommerceCurrency(
+					currencyId);
+
+			if (commerceCurrency != null) {
+				return commerceCurrency;
+			}
+		}
+
+		if (Validator.isNull(currencyExternalReferenceCode)) {
+			commerceCurrency =
+				CommerceCurrencyLocalServiceUtil.
+					fetchCommerceCurrencyByExternalReferenceCode(
+						currencyExternalReferenceCode, companyId);
+
```

Comment:
@luca-pellizzon , Is this and line 25 intentional? Wouldn't `commerceCurrency` always be null with these checks? We are passing in null `currencyCode` and null `currencyExternalReferenceCode`

Reply:
must be a isNotNull, thanks :+1:

Reply:
Besides this, everything else looks good to me.